### PR TITLE
#39 remove css and add less files

### PR DESCRIPTION
**Pull request by ndufreche**
:warning: this PR is more large than the issue 39

This PR remove all css in our current version.
All CSS dependencies are managed with bower.

I have clean bower and npm dependencies too.

And there is a new Grunt command available to build css files :
```
grunt less:css
```

### CODE REVIEW

**Comment by ndufreche** [comment]
The result css is a little bugged, du on old bad practice. The css have to be corrected in another issue

**Comment by eric-chau** [comment]
It seems to be really good, :+1: well done.

File: Gruntfile.js [hunk 1]
```diff
@@ -226,21 +202,21 @@ module.exports = function (grunt) {
...
+
     // grunt tasks
-    grunt.registerTask('default', ['bower', 'jshint', 'jslint', 'jasmine:test', 'concat', 'uglify']);
-    grunt.registerTask('test', ['bower', 'jshint', 'jslint', 'jasmine:coverage']);
-    grunt.registerTask('build', ['bower', 'concat', 'uglify']);
-    grunt.registerTask('jenkins', ['bower', 'jshint', 'jslint', 'jasmine:coverage', 'concat', 'uglify']);
+    grunt.registerTask('default', ['less:css', 'jshint', 'jslint', 'jasmine:coverage', 'concat', 'uglify']);
+    grunt.registerTask('test', ['jshint', 'jslint', 'jasmine:coverage']);
+    grunt.registerTask('build', ['less:css', 'bower', 'concat', 'uglify']);
+    grunt.registerTask('jenkins', ['jshint', 'jslint', 'jasmine:coverage', 'concat', 'uglify']);
```
Comment (mickaelandrieu) [inline Gruntfile.js]: do we need jenkins ? maybe rename it to "travis" ?

File: bower.json [hunk 2]
```diff
@@ -14,12 +14,11 @@
   "private": true,
   "dependencies": {
     "jquery": "~2.1.1",
-    "handlebars": "~2.0.0",
     "requirejs": "~2.1.14",
     "requirejs-text": "~2.0.12",
     "backbone": "~1.1.2",
     "cryptojslib": "*",
-    "bootstrap": "*",
+    "bootstrap": "3.1.0",
```
Comment (mickaelandrieu) [inline bower.json]: bootstrap is 3.3.2, you should use ``~3.3`` notation.

Comment (ndufreche) [inline bower.json]: The actual BB less is only compatible with this particular version.

File: Gruntfile.js (hunk 1 above)
Comment (ndufreche) [inline Gruntfile.js]: Travis use the grunt test command.
I remove this line

File: less/bb5-ui-login.less [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    & * {
+        #bb5-ui .bb5-ui-reset;
+    }
+
+	.ui-dialog {
+		#bb5-ui .ui-widget;
+	}
+
+	.ui-widget-header {
+        #bb5-ui .ui-widget-header;
```
Comment (mickaelandrieu) [inline less/bb5-ui-login.less]: maybe an indentation issue :)

File: less/bb5-ui.less [hunk 4]
```diff
@@ -0,0 +1,2452 @@
...
+    @import "@{font-awesome-path}less/list.less";
+    @import "@{font-awesome-path}less/bordered-pulled.less";
+    @import "@{font-awesome-path}less/spinning.less";
+    @import "@{font-awesome-path}less/rotated-flipped.less";
+    @import "@{font-awesome-path}less/stacked.less";
+    @import "@{font-awesome-path}less/icons.less";
+}
+
+/*import bb5 vars*/
+//@import "bb5-var.less";
```
Comment (mickaelandrieu) [inline less/bb5-ui.less]: to be removed, right ?

File: less/bb5-ui.less [hunk 5]
```diff
@@ -0,0 +1,2452 @@
...
+    z-index:490000;
+    top:0;
+    left:0;
+    right:0;
+    min-height:192px;
+
+    &.bb5-hidden {
+        display:none;
+    }
+    /**/
```
Comment (mickaelandrieu) [inline less/bb5-ui.less]: to be removed too :)

File: less/bb5-ui.less [hunk 6]
```diff
@@ -0,0 +1,2452 @@
...
+        display: table-cell;
+        vertical-align: middle;
+    }
+
+    .bb5-border {
+        border:1px solid @bb5-border-color;
+        padding:14px 7px;
+    }
+
+    .bb5-border-radius {
```
Comment (mickaelandrieu) [inline less/bb5-ui.less]: to be removed or uncommented ?

File: less/bb5-ui.less [hunk 7]
```diff
@@ -0,0 +1,2452 @@
...
+    }
+
+    .bb5-form-wrapper {
+        .bb5-border;
+    }
+
+    textarea.form-control {
+        height: auto;
+    }
+    /**/
```
Comment (mickaelandrieu) [inline less/bb5-ui.less]: remove all the ``/**/`` please :)

File: less/bb5-ui.less [hunk 8]
```diff
@@ -0,0 +1,2452 @@
...
+    .btn-sm {
+        font-size:11px;
+        padding:3px 6px 4px 6px;
+    }
+
+    .btn-xs {
+        font-size:11px;
+    }
+
+    /*.btn-xs:before(.btn-transparent) .fa {margin-left: -6px;}*/
```
Comment (mickaelandrieu) [inline less/bb5-ui.less]: to be removed.

File: less/bb5-ui.less [hunk 9]
```diff
@@ -0,0 +1,2452 @@
...
+        }
+    }
+    /**/
+    .bb5-theme-bg-preview {
+        width: 107px;
+        height: 107px;
+    }
+
+    /*BOX*/
+    .bb5-box-data {
```
Comment (mickaelandrieu) [inline less/bb5-ui.less]: ? is it used by application ?

File: less/bb5-ui.less [hunk 10]
```diff
@@ -0,0 +1,2452 @@
...
+
+            .jqtree-element {
+                position: relative;
+            }
+
+            li.jqtree-selected > .jqtree-element {
+
+                &,
+                &:hover {
+                    background: rgb(255,255,255); /* Old browsers */
```
Comment (mickaelandrieu) [inline less/bb5-ui.less]:
```css
background: #FFF;
```

File: bower.json (hunk 2 above)
Comment (mickaelandrieu) [inline bower.json]: :'(

**Comment by mickaelandrieu** [comment]
Anyway, this is a great enhancement :+1:

File: less/bb5-ui.less (hunk 6 above)
Comment (ndufreche) [inline less/bb5-ui.less]: I don't have any idea

File: less/bb5-ui.less (hunk 9 above)
Comment (ndufreche) [inline less/bb5-ui.less]: I don't have any idea

**Pull request merged**